### PR TITLE
add driver test command to cirrus

### PR DESCRIPTION
**Pull request by collinjackson**
Fixes flutter/flutter#26080

This teaches Cirrus to run the new drive-examples command added in flutter/plugin_tools#29 on iOS and Android simulators.

I'm sharding the plugins tests more and requesting more powerful Linux VMs to ensure that the tests don't time out now that they're running simulators.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -22,11 +22,18 @@ task:
...
     - name: analyze
       script: ./script/incremental_build.sh analyze
-    - name: build-apks+java-test
+    - name: build-apks+java-test+drive-examples
       env:
         matrix:
           PLUGIN_SHARDING: "--shardIndex 0 --shardCount 2"
           PLUGIN_SHARDING: "--shardIndex 1 --shardCount 2"
+      create_device_script:
+        echo no | avdmanager create avd --force \
```
Comment (amirh) [inline .cirrus.yml]: Is this slow? is so maybe worth keeping it as part of the docker image?

Comment (collinjackson) [reply .cirrus.yml]: According to https://medium.com/cirruslabs/speeding-up-android-builds-with-cirrus-ci-9e20ab7c5150 it's pretty fast but also I don't see a downside to putting it in the Dockerfile so I'll try that

Comment (amirh) [reply .cirrus.yml]: If it's fast probably simpler to keep it here

File: packages/shared_preferences/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+[test change to make bots run the tests]
```
Comment (collinjackson) [inline packages/shared_preferences/CHANGELOG.md]: will be reverted prior to merging

File: .cirrus.yml [hunk 3]
```diff
@@ -22,11 +22,18 @@ task:
       test_script: ./script/incremental_build.sh test
     - name: analyze
       script: ./script/incremental_build.sh analyze
-    - name: build-apks+java-test
+    - name: build-apks+java-test+drive-examples
       env:
         matrix:
           PLUGIN_SHARDING: "--shardIndex 0 --shardCount 2"
           PLUGIN_SHARDING: "--shardIndex 1 --shardCount 2"
+          PLUGIN_SHARDING: "--shardIndex 2 --shardCount 2"
```
Comment (amirh) [inline .cirrus.yml]: should shardCount say 4?

Comment (collinjackson) [reply .cirrus.yml]: good catch, I'll see if I can get the tests to pass with just 2 shards and if not increase to 4

**Pull request merged**